tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline/taxi_pipeline_dummy_e2e_test.py

```diff
@@ -0,0 +1,155 @@
+    self._module_file = os.path.join(taxi_root, 'taxi_utils.py')
+    self._serving_model_dir = os.path.join(self._test_dir, 'serving_model')
+    self._pipeline_root = os.path.join(self._test_dir, 'tfx', 'pipelines',
+                                       self._pipeline_name)
+    self._metadata_path = os.path.join(self._test_dir, 'tfx', 'metadata',
+                                       self._pipeline_name, 'metadata.db')
+
+  def assertExecutedOnce(self, component: Text) -> None:
+    """Check the component is executed exactly once."""
+    component_path = os.path.join(self._pipeline_root, component)
+    self.assertTrue(tf.io.gfile.exists(component_path))
+    outputs = tf.io.gfile.listdir(component_path)
+    for output in outputs:
+      execution = tf.io.gfile.listdir(os.path.join(component_path, output))
+      self.assertEqual(1, len(execution))
+
+  def assertPipelineExecution(self) -> None:
+    self.assertExecutedOnce('CsvExampleGen')
+    self.assertExecutedOnce('Evaluator')
+    self.assertExecutedOnce('ExampleValidator')
+    self.assertExecutedOnce('Pusher')
+    self.assertExecutedOnce('SchemaGen')
+    self.assertExecutedOnce('StatisticsGen')
+    self.assertExecutedOnce('Trainer')
+    self.assertExecutedOnce('Transform')
+
+  def testTaxiPipelineBeam(self):
+    record_dir = os.path.join(os.environ['HOME'],
+                              'tfx/tfx/experimental/pipeline_testing/',
+                              'examples/chicago_taxi_pipeline/testdata')
+
+    component_ids = ['CsvExampleGen', \
+                    'StatisticsGen', 'SchemaGen', \
+                    'ExampleValidator', 'Transform', \
+                    'Trainer', 'Evaluator', 'Pusher']
+
+    MyDummyLauncher = \
+      dummy_component_launcher.create_dummy_launcher_class(record_dir,
+                                                           component_ids,
+                                                           {})
+    BeamDagRunner(config=pipeline_config.PipelineConfig(
+        supported_launcher_classes=[
+            MyDummyLauncher,
+        ],
+        )).run(
+            taxi_pipeline_regression_test._create_pipeline(  # pylint: disable=protected-access
+                pipeline_name=self._pipeline_name,
+                data_root=self._data_root,
+                module_file=self._module_file,
+                serving_model_dir=self._serving_model_dir,
+                pipeline_root=self._pipeline_root,
+                metadata_path=self._metadata_path,
+                direct_num_workers=1))
+
+    self.assertTrue(tf.io.gfile.exists(self._metadata_path))
+    metadata_config = metadata.sqlite_metadata_connection_config(
+        self._metadata_path)
+    with metadata.Metadata(metadata_config) as m:
+      artifact_count = len(m.store.get_artifacts())
+      execution_count = len(m.store.get_executions())
+      self.assertGreaterEqual(artifact_count, execution_count)
+      self.assertEqual(9, execution_count)
```

Comment:
Can you change this hard coded number with computation from other source? For example `len(some_pipeline.components)`?